packages/components/src/components/files/FileTree.tsx

```diff
@@ -0,0 +1,388 @@
+import { Treebeard, decorators } from 'react-treebeard';
+
+import React, { PureComponent } from 'react';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faFolder, faFileAlt } from '@fortawesome/free-solid-svg-icons';
+import { Checkbox, Alert } from 'react-bootstrap';
+import { List } from 'immutable';
+import {LoadingSpinner} from "../..";
+
+const customStyle = {
+    tree: {
+        base: {
+            listStyle: 'none',
+            backgroundColor: 'white',
+            margin: 0,
+            padding: 0,
+            color: '#777',
+            fontFamily: 'lucida grande ,tahoma,verdana,arial,sans-serif',
+            fontSize: '14px',
+        },
+        node: {
+            base: {
+                position: 'relative',
+            },
+            link: {
+                cursor: 'pointer',
+                position: 'relative',
+                padding: '0px 5px',
+                display: 'block',
+            },
+            activeLink: {
+                background: 'white',
+            },
+            toggle: {
+                base: {
+                    position: 'relative',
+                    display: 'inline-block',
+                    verticalAlign: 'top',
+                    marginLeft: '-5px',
+                    height: '24px',
+                    width: '24px',
+                },
+                wrapper: {
+                    position: 'absolute',
+                    top: '50%',
+                    left: '50%',
+                    margin: '-7px 0 0 -7px',
+                    height: '14px',
+                },
+                height: 14,
+                width: 14,
+                arrow: {
+                    fill: '#777',
+                    strokeWidth: 0,
+                },
+            },
+            header: {
+                base: {
+                    display: 'inline-block',
+                    verticalAlign: 'top',
+                    color: '#777',
+                },
+                connector: {
+                    width: '2px',
+                    height: '12px',
+                    borderLeft: 'solid 2px black',
+                    borderBottom: 'solid 2px black',
+                    position: 'absolute',
+                    top: '0px',
+                    left: '-21px',
+                },
+                title: {
+                    lineHeight: '24px',
+                    verticalAlign: 'middle',
+                },
+            },
+            subtree: {
+                listStyle: 'none',
+                paddingLeft: '19px',
+            },
+            loading: {
+                color: '#777',
+            },
+        },
+    },
+};
+
+const DEFAULT_ROOT_PREFIX = '|root';
+const CHECK_ID_PREFIX = 'filetree-check-';
+const EMPTY_FILE_NAME = '*empty';
+const LOADING_FILE_NAME = '*loading';
+
+const Header = (props): React.ReactFragment => {
```

Comment:
I'm confused why this isn't actually a component? It could even just be a function component, doesn't have to extend Component/PureComponent, but for some reason it's not even that? I am confused. I will email you a patch to show you what I am talking about.